packages/lesswrong/components/messaging/InboxWrapper.tsx

```diff
@@ -12,7 +12,7 @@ export type InboxComponentProps = {
   classes: ClassesType;
 };
 
-const InboxWrapper = () => {
+const InboxWrapper = ({friendlyInbox = isFriendlyUI}: { friendlyInbox: boolean }) => {
```

Comment:
Could you explain what the purpose of this is? The sites are already classified as `isFriendlyUI` or not, and once LW switches over to this new inbox then I assume we won't need to support the unfriendly inbox anymore

Reply:
this just elevates a check this component had internally to be a prop

the point of this is to be able to specify a coherent behaviour in consistent manner. current codebase default (friendly UI without overrides) would give you "friendly" all messages view and old individual message view. part of the reason for that discrepancy is that logic on what ui to show is spread between routes and inbox wrapper components. (you can see me being confused about it earlier in this PR)
this change makes routes fully responsible for selecting what components we're showing.

Reply:
(ideally we just switch to friendly inbox now as you previously suggested, but @lagerros never replied to my messages about it, so I figured I'd just ensure a coherent behaviour matching current configurations)

Reply:
Sorry I don't think I understand. In `routes.ts` you've already made an update such that the default is to use `InboxWrapper` for both the all messages view and the individual message view, and for LW/AF there is a special case where they use `InboxWrapper` for the all messages view and `ConversationWrapper` for the individual message view. `InboxWrapper` was already checking `isFriendlyUI`, so it should show the new inbox for all sites except for LW/AF. That should already accomplish what we want right? We don't need these additional `componentProps` passed in?

Reply:
you are correct. 

this occurred to me only I've implemented the current approach 😛. I figured I'd keep it as it's more transparent 
(I was imagining people hitting inconsistency if they were to theoretically start a new book ui forum and want new inbox. Now it's clearer how to get there)

but if you say "this adds more complexity for the hypothetical case we won't have" that's fair, and I can remove the props stuff

Reply:
Yeah I generally prefer not to add unused code to the codebase, and in this case I expect all forums to end up using the nice new inbox components anyway. Could you remove the `componentProps` bits?

Reply:
will do

Reply:
updated the pr!